docs/en/02_Developer_Guides/14_Files/05_File_Migration.md

```diff
@@ -114,6 +114,121 @@ Use the following estimates to decide how you will run your file migration:
+
+SilverStripe 3 did not support draft file. So all migrated SilverStripe 3 files will automatically be published by the File Migration Task.
+
+SilverStripe 4 has different naming convention. File names can not contain "double underscores" in SilverStripe 4 (`__`) because those characters are used to indicate a "file variant". SilverStripe 3 files not conforming to this rule will automatically be renamed by the file migration task.
+
+### Will old URLs to SilverStripe 3 files still resolve?
+
+In most cases, the SilverStripe 3 file path will be identical to the public path of the file in SilverStripe 4. Links to SilverStripe 3 file variants (e.g.: cropped images) will automatically be redirected to their new location.
+
+### What about file references in `HTMLText` database fields?
+
+The short code format in SilverStripe 4 has changed from SilverStripe 3. This means that when CMS editors try to edit existing content and alter image or file links, the reference to the file will not work.
+
+After running the file migration, you can run a the short code migration task to update your existing HTML content in your database.
+
+```bash
+sake dev/tasks/TagsToShortcode
+``` 
+
+This will rewrite your existing shortcode to the newer format SilverStripe 4 expects.
+
+## Migrating from SilverStripe 4.3 or below to SilverStripe 4.4 or above
+
+Silverstripe 4.0 to 4.3 were using hash paths for the public file store as well as the protected file store. This meant that every time a new version of a file was published, it would be served under a different URL.
+
+SilverStripe 4.4 and later default to using natural paths for public store files. After upgrading to SilverStripe 4.4 or above, you can run the file migration task to normalise your existing files.
+
+This will take all existing files under a hash path and move them to their natural path.
+
+### Do I need to re-migrate my files?
+
+No, you do not need to run the file migration task when upgrading to SilverStripe 4.4 or later. 
+
+Your existing files will still resolve. Old hash paths will be redirected to the latest version of the file. As newer versions of existing files are published, they will be moved to their natural path.
+
+You can choose to run the migration task at any point.
+
+### What if I’m using `legacy_filenames`?
+
+SilverStripe 4.0 to 4.3 allowed you to enable a `legacy_filenames` mode. This would be equivalent to using natural path for both the public store and the protected store.
+
+With the introduction of natural path for the public store, the primary reason for `legacy_filenames` mode is no longer relevant. `legacy_filenames` mode is now considered deprecated, however leaving it enabled will not have any impact on your project – it will simply be ignored.
```

Comment:
```suggestion
With the introduction of natural paths for the public store, the primary reason for `legacy_filenames` mode is no longer relevant. `legacy_filenames` mode is now considered deprecated, however leaving it enabled will not have any impact on your project – it will simply be ignored.
```